docs/internal/GeneralArchitectureGuide.md

```diff
@@ -68,6 +68,43 @@ state must ever be reloaded from persisted state.
 
 ## Deprecations
 
+## Backwards Compatibility
+
+Elasticsearch has major versions (e.g. 8.0.0), minor versions (e.g. 8.3.0) and patch versions (e.g. 8.3.1). Multiversion
+binary backwards compatibility within a cluster is guaranteed for specific situations.
+
+### Transport Layer Backwards Compatibility
+
+Elasticsearch nodes are backwards compatible with all earlier binary version nodes in the same major release: all the earlier
+minor and patch releases within that same major version. All binary versions within one major version X are also compatible with
+the last minor release version of the previous major version, e.g. (X-1).last.0. More concretely, all 8.x.x versions are
+backwards compatible with 7.17.x versions.
+
+### Index Format Backwards Compatibility
+
+Index format backwards compatibility is guaranteed with the entirety of the previous major release version. All 8.x.x versions,
+for example, are backwards compatible with all 7.x.x versions. 9.x.x versions, however, will not be compatible with 7.x.x format
```

Comment:
Again "are backwards compatible" is too broad.

Reply:
Rephrased.